### PR TITLE
Introduce authentication for zmq communications

### PR DESCRIPTION
**Pull request by mraspaud**
This PR aims at introducing authentication in the posttroll communications.

- [x] Create a zmq backend
- [x] Extract zmq-specific components to the backend
- [x] Create secure zmq alternatives to the zmq components

### CODE REVIEW

File: posttroll/subscriber.py [hunk 1]
```diff
@@ -455,28 +340,29 @@ def create_subscriber_from_dict_config(settings):
...
-    message_filter = settings.get('message_filter', None)
-    translate = settings.get('translate', False)
+    _ = settings.pop("nameserver", None)
+    _ = settings.pop("port", None)
+    _ = settings.pop("services", None)
+    _ = settings.pop("addr_listener", None),
+    _ = settings.pop("timeout", None)
 
-    return Subscriber(addresses, topics=topics, message_filter=message_filter, translate=translate)
+    return Subscriber(**settings)
```
Comment (pnuu) [inline posttroll/subscriber.py:355]: I think the original is cleaner as well as explicit. If there are extra items in `settings` the `Subscriber` creation will fail.

**Review by pnuu — COMMENTED** [review]
First of all, great work with the refactoring!

Could you run `flake8` and see if there are manageable amount of things to fix? There were some missing new-lines in at least `message.py`.

Couple additional comments inline.

File: posttroll/subscriber.py (hunk 1 above)
Comment (mraspaud) [reply posttroll/subscriber.py:355]: Yes, that's the point. Until now we could pass whatever and not even know it was wrong...

File: posttroll/ns.py [hunk 2]
```diff
@@ -23,35 +23,36 @@
...
-    """A timeout."""
+def get_configured_nameserver_port():
+    """Get the configured nameserver port."""
+    try:
+        port = int(os.environ["NAMESERVER_PORT"])
+        warnings.warn("NAMESERVER_PORT is pending deprecation, please use POSTTROLL_NAMESERVER_PORT instead.",
+                    PendingDeprecationWarning)
+    except KeyError:
+        port = DEFAULT_NAMESERVER_PORT
+    return config.get("nameserver_port", port)
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply posttroll/ns.py:54]: yes, that's Donfig

File: posttroll/address_receiver.py [hunk 3]
```diff
@@ -35,27 +35,28 @@
...
-default_publish_port = 16543
+DEFAULT_ADDRESS_PUBLISH_PORT = 16543
 
 ten_minutes = dt.timedelta(minutes=10)
 zero_seconds = dt.timedelta(seconds=0)
 
 
+def get_configured_address_port():
+    return config.get("address_publish_port", DEFAULT_ADDRESS_PUBLISH_PORT)
+
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply posttroll/address_receiver.py:59]: Using ruff now, which can't count empty lines I think

[the remark this answers is not in the document]
Comment (mraspaud) [reply posttroll/address_receiver.py:59]: Ok, should be fixed now

[the remark this answers is not in the document]
Comment (mraspaud) [reply posttroll/address_receiver.py:59]: For reference, the `--preview` flag needs to be added to ruff for the blank lines.

**Pull request merged**